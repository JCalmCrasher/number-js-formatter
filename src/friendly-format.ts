import lookup from "./number-lookup";

export function friendlyFormat(number: number, digits: number = 0) {
    const lookupNumbers = lookup;

    const regex = /\.0+$|(\.[0-9]*[1-9])0+$/;

    let result = lookupNumbers.slice().reverse().find(function (item) {
        return number >= item.value;
    });

    return result ? (number / result.value).toFixed(digits).replace(regex, "$1") + result.symbol : "0";
}