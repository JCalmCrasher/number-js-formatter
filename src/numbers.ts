export const THOUSAND = 1000;
export const MILLION = 1000_000;
export const BILLION = 1000_000_000;
export const TRILLION = 1000_000_000_000;
