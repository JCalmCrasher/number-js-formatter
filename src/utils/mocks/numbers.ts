export const THOUSAND = 1000;
export const MILLION = 1000_000;
export const BILLION = 1000_000_000;
export const TRILLION = 1000_000_000_000;

export const THOUSAND_FLOAT = 1010.014;
export const MILLION_FLOAT = 1_923_000.156;
export const BILLION_FLOAT = 1_034_050_000.91332;
export const TRILLION_FLOAT = 1105_830_000_302.9946;
