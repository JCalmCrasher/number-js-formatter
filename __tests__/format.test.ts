import { friendlyFormat } from '@/friendly-format';
import { BILLION, MILLION, THOUSAND, TRILLION } from '@/numbers';
import { describe, expect, it } from 'vitest';

describe('format', () => {
  it('should format thousand', () => {
    expect(friendlyFormat(THOUSAND)).toMatchSnapshot();
  });

  it('should format million', () => {
    expect(friendlyFormat(MILLION)).toMatchSnapshot();
  });

  it('should format billion', () => {
    expect(friendlyFormat(BILLION)).toMatchSnapshot();
  });

  it('should format trillion', () => {
    expect(friendlyFormat(TRILLION)).toMatchSnapshot();
  });
});
