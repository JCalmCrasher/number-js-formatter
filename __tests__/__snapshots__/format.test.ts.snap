// Vitest Snapshot v1

exports[`format > should format billion 1`] = `"1bn"`;

exports[`format > should format million 1`] = `"1m"`;

exports[`format > should format thousand 1`] = `"1k"`;

exports[`format > should format trillion 1`] = `"1tn"`;
